x-pack/plugins/stack_connectors/server/connector_types/inference/index.ts

```diff
@@ -0,0 +1,182 @@
+  ValidatorType,
+} from '@kbn/actions-plugin/server/sub_action_framework/types';
+import {
+  GenerativeAIForSearchPlaygroundConnectorFeatureId,
+  GenerativeAIForSecurityConnectorFeatureId,
+} from '@kbn/actions-plugin/common';
+import { ValidatorServices } from '@kbn/actions-plugin/server/types';
+import { GenerativeAIForObservabilityConnectorFeatureId } from '@kbn/actions-plugin/common/connector_feature_config';
+import { InferenceTaskType } from '@elastic/elasticsearch/lib/api/types';
+import { ElasticsearchClient, Logger } from '@kbn/core/server';
+import {
+  INFERENCE_CONNECTOR_TITLE,
+  INFERENCE_CONNECTOR_ID,
+  ServiceProviderKeys,
+  SUB_ACTION,
+} from '../../../common/inference/constants';
+import { ConfigSchema, SecretsSchema } from '../../../common/inference/schema';
+import { Config, Secrets } from '../../../common/inference/types';
+import { InferenceConnector } from './inference';
+import { unflattenObject } from '../lib/unflatten_object';
+
+const deleteInferenceEndpoint = async (
+  inferenceId: string,
+  taskType: InferenceTaskType,
+  logger: Logger,
+  esClient: ElasticsearchClient
+) => {
+  try {
+    await esClient.inference.delete({
+      task_type: taskType,
+      inference_id: inferenceId,
+    });
+    logger.info(
+      `Inference endpoint for task type "${taskType}" and inference id ${inferenceId} was successfuly deleted`
+    );
+  } catch (e) {
+    logger.warn(
+      `Failed to delete inference endpoint for task type "${taskType}" and inference id ${inferenceId}. Error: ${e.message}`
+    );
+    throw e;
+  }
+};
+
+export const getConnectorType = (): SubActionConnectorType<Config, Secrets> => ({
+  id: INFERENCE_CONNECTOR_ID,
+  name: INFERENCE_CONNECTOR_TITLE,
+  getService: (params) => new InferenceConnector(params),
+  schema: {
+    config: ConfigSchema,
+    secrets: SecretsSchema,
+  },
+  validators: [{ type: ValidatorType.CONFIG, validator: configValidator }],
+  supportedFeatureIds: [
+    GenerativeAIForSecurityConnectorFeatureId,
+    GenerativeAIForSearchPlaygroundConnectorFeatureId,
+    GenerativeAIForObservabilityConnectorFeatureId,
+  ],
+  minimumLicenseRequired: 'enterprise' as const,
+  preSaveHook: async ({ config, secrets, logger, services, isUpdate }) => {
+    const esClient = services.scopedClusterClient.asInternalUser;
+    try {
+      const taskSettings = config?.taskTypeConfig
+        ? {
+            ...unflattenObject(config?.taskTypeConfig),
+          }
+        : {};
+      const serviceSettings = {
+        ...unflattenObject(config?.providerConfig ?? {}),
+        ...unflattenObject(secrets?.providerSecrets ?? {}),
+      };
+
+      let inferenceExists = false;
+      try {
+        await esClient?.inference.get({
+          inference_id: config?.inferenceId,
+          task_type: config?.taskType as InferenceTaskType,
+        });
+        inferenceExists = true;
+      } catch (e) {
+        /* throws error if inference endpoint by id does not exist */
```

Comment:
There are other reasons why an error could be thrown, even if it does exist.  Probably would be good to specifically check if this is a "not found" error (we do this somewhere in alerting, we can look up the magic incantation) vs something else, and if it's something else, throw THAT error.